connectors/src/connectors/confluence/lib/hierarchy.ts

```diff
@@ -0,0 +1,63 @@
+import type { ModelId } from "@dust-tt/types";
+
+import {
+  makeConfluenceInternalPageId,
+  makeConfluenceInternalSpaceId,
+} from "@connectors/connectors/confluence/lib/internal_ids";
+import { ConfluencePage } from "@connectors/lib/models/confluence";
+
+interface RawConfluencePage {
+  pageId: string;
+  parentId: string | null;
+  spaceId: string;
+}
+
+export async function getConfluencePageParentIds(
```

Comment:
Wondering about complexity here; for each page, a query retrieving all pages is sent?
If yes, worth caching the "all pages" which is always the same across all calls? (redis not needed though)

Reply:
See comments at call site, it's coming in another PR 🙃.

Reply:
Addressed here https://github.com/dust-tt/dust/pull/3468.